lib/codecs/src/encoding/format/csv.rs

```diff
@@ -181,4 +238,77 @@ mod tests {
             b"value1,value5,value5,value3,value2".as_slice()
         );
     }
+
+    #[test]
+    fn correct_quoting() {
+        let event = Event::Log(LogEvent::from(btreemap! {
+            "field1" => Value::from("value1 \" value2"),
+        }));
+        let fields = vec![
+            ConfigTargetPath::try_from("field1".to_string()).unwrap(),
+        ];
+        let mut opts = CsvSerializerOptions::default();
+        opts.fields = fields;
+
+        let config = CsvSerializerConfig::new(opts);
+        let mut serializer = config.build().unwrap();
+        let mut bytes = BytesMut::new();
+        serializer.encode(event, &mut bytes).unwrap();
+
+        assert_eq!(
+            bytes.freeze(),
+            b"\"value1 \"\" value2\"".as_slice()
+        );
+    }
+
+    #[test]
+    fn custom_delimiter() {
```

Comment:
Thank you for providing these tests!

If `custom_delimiter` fails due to https://github.com/vectordotdev/vector/issues/17261 we can (1) make sure this test passes and (2) make a note on current behavior vs desired behavior.

Reply:
From my point of view it is the other way around. Writing `custom_escape_char` I could not get it to work. Digging into that I wrote `custom_delimiter` as well as found out about [this issue](https://github.com/BurntSushi/rust-csv/issues/331) which I think is a bug in the current version of vector?